### DISCUSSION TITLE
Why can't I preview images via HTTP?

### DISCUSSION THREAD

**Discussion by ChangqingChen** [coder/code-server#4572] [Q&A]
Why can't I preview images and markdown via http? Use https is normal. 
![image](https://user-images.githubusercontent.com/19241013/144193922-2c9ced88-c102-4eb8-82d2-009dbe0774d6.png)

**Comment by jsjoeio** [comment]
I believe this is due to how images and markdown is loaded inside VS Code. IIRC, they use a webview. The webview talks to a service worker which handles serving image assets and rendering the markdown preview. That means if you're using HTTP (or incognito I believe in Firefox) then service workers will be disabled (by the browser), which means your resources won't load. 

Hope that helps 👍

**Reply by ChangqingChen** [reply]
Is there any way to solve this problem?  coder-server is deployed in LAN. I can only communicate through HTTP now.

**Reply by ChangqingChen** [reply]
Is it possible to disable service worker at code-server?

**Reply by jsjoeio** [reply]
> I can only communicate through HTTP now

Could you use something like `ngrok` or `localtunnel` and access your instance through that to get HTTPS?

> Is it possible to disable service worker at code-server?

Hmm...I don't think so since VS Code uses them under the hood.

**Reply by ChangqingChen** [reply]
I don't think the colleague will allow me to do this. Is there a way to load resources correctly without using HTTPS

**Reply by ChangqingChen** [reply]
can't preview images, many extensions won't work

**Reply by jsjoeio** [reply]
Unfortunately, I don't think the browsers will allow it :( Preview images and many extensions rely on service workers and service workers won't work without HTTPS.

Can you explain the situation to your colleague?
